packages/webapp/src/_app/eos/secret-publisher.ts

```diff
@@ -0,0 +1,252 @@
+import { PrivateKey, PublicKey } from "eosjs/dist/eosjs-jssig";
+import { generateKeyPair } from "eosjs/dist/eosjs-key-conversions";
+import { KeyType } from "eosjs/dist/eosjs-numeric";
+
+/**
+ * Provides a way to encrypt data to be published on chain. Useful for
+ * secretly sharing any temporal data. Eg. induction meeting links, election
+ * room links.
+ *
+ * IMPORTANT NOTES:
+ * - Any data on chain will eventually be decrypted. Never use it for PII (read the
```

Comment:
If encrypted data were not decrypted, there would be no point to storing it at all.  With that being said, the level of security is not suitable for long-term protection of high-value secrets.

Reply:
Yeah here I meant more of decrypted by badactors/attackers. I will rephrase and add your point.